From b17a6448e31684225e8c28aeba661a71a22dc917 Mon Sep 17 00:00:00 2001
Message-Id: <b17a6448e31684225e8c28aeba661a71a22dc917@dist-git>
From: Peter Krempa <pkrempa@redhat.com>
Date: Thu, 5 Jan 2017 09:48:13 +0100
Subject: [PATCH] qemu: snapshot: Resume VM after live snapshot

Commit 4b951d1e38259ff5d03e9eedb65095eead8099e1 missed the fact that the
VM needs to be resumed after a live external checkpoint (memory
snapshot) where the cpus would be paused by the migration rather than
libvirt.

(cherry picked from commit 2e86c0816fc8ab573745f1a9a650be09bd66e300)

https://bugzilla.redhat.com/show_bug.cgi?id=1406765
---
 src/qemu/qemu_driver.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index 2e442a987..37b09998d 100644
--- a/src/qemu/qemu_driver.c
+++ b/src/qemu/qemu_driver.c
@@ -14334,6 +14334,9 @@ qemuDomainSnapshotCreateActiveExternal(virConnectPtr conn,
          * atomic flag now says whether we need to pause, and a
          * capability bit says whether to use transaction.
          */
+        if (memory)
+            resume = true;
+
         if ((memory && !(flags & VIR_DOMAIN_SNAPSHOT_CREATE_LIVE)) ||
             (!memory && atomic && !transaction)) {
             if (qemuProcessStopCPUs(driver, vm, VIR_DOMAIN_PAUSED_SNAPSHOT,
-- 
2.11.0

